greg/aux_functions.py

```diff
@@ -176,7 +176,21 @@ def tag(placeholders):
     filename = substitute_placeholders(template, placeholders)
     podpath = os.path.join(placeholders.directory, filename)
     # ... and this is it
-
+    # We also retrieve the path of a cover image, if there is one
+    coverart = placeholders.feed.retrieve_config("coverart", False)
+    if coverart:
+        import mimetypes
+        coverart_filename = substitute_placeholders(coverart, placeholders)
+        if not os.path.exists(coverart_filename):
+            print("""The file that I was supposed to use as cover art does not
+                    exist.""", file=sys.stderr, flush=True)
+            coverart = False
+        else:
+            coverart_mime = mimetypes.guess_type(coverart_filename)[0]
```

Comment:
Rather than guessing the MIME type from the extension, couldn’t we leverage mime-magic on systems that have it? It’s more accurate in most cases.

Reply:
E.g. using https://github.com/ahupp/python-magic

Reply:
I am a bit wary of multiplying dependencies; also, users can pretty much choose their own extension with `file_to_tag`, right?

Reply:
I had assumed that EyeD3 bundled magic anyway for use with https://eyed3.readthedocs.io/en/latest/eyed3.plugins.html#module-eyed3.plugins.mimetype but looking at the source, it seems I was mistaken.

Reply:
... but perhaps I'm being silly about dependencies. You seem to be a serious dev, much unlike me :) So I'm willing to listen.

Reply:
No, I think you’re being perfectly reasonable. Also I might be a serious dev (whatever that means) but not all that familiar with the Python ecosystem…